ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/ActiveContractKeysPoolSpec.scala

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.v1.value.Value
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class ActiveContractKeysPoolSpec extends AnyFlatSpec with Matchers {
+
+  it should "put and pop from a pool" in {
+    val tested = new ActiveContractKeysPool(RandomnessProvider.forSeed(0))
+    intercept[NoSuchElementException](tested.getAndRemoveContractKey(templateName = "t1"))
+    tested.addContractKey(templateName = "t1", key = makeValue("1"))
+    intercept[NoSuchElementException](tested.getAndRemoveContractKey(templateName = "t2"))
+    tested.getAndRemoveContractKey("t1") shouldBe makeValue("1")
+    intercept[IndexOutOfBoundsException](tested.getAndRemoveContractKey(templateName = "t1"))
```

Comment:
This this interface was being used more widely I would push this call to return an option.